src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -137,10 +138,30 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
   private var database: Database = _
   private var collectionThroughput: Option[Int] = None
 
+  private val rnd = scala.util.Random
+
+  private val refreshDelay : Long = (10 * 60 * 1000) + rnd.nextInt(5 * 60 * 1000) // in 10 - 15 minutes
+  private val refreshPeriod : Long = 15 * 60 * 1000 // every 15 minutes
+
+  // main purpose of the time is to allow bulk operations to consume
+  // additional throughput when more RUs are getting provisioned
+  private val timerName = "collectionthroughput-refresh-timer"
+  private val timer: Timer = new Timer(timerName)
+
   @transient private var bulkImporter: DocumentBulkExecutor = _
 
   private var documentClient: DocumentClient = CosmosDBConnection.getClient(connectionMode, getClientConfiguration(config))
 
+  logInfo(s"${timerName}: scheduling timer - delay: ${refreshDelay} ms, period: ${refreshPeriod} ms")
+  timer.schedule(
```

Comment:
Since case classes are serialized and sent across the wire back and forth, I think this timer is probably not firing the way you expect. When the objects gets rehydrated, the Timer and the intervals would be new'd up again.